Sources/DatadogObjc/Tracer+objc.swift

```diff
@@ -11,7 +11,8 @@ import struct Datadog.OTReference
 import class Datadog.HTTPHeadersWriter
 
 @objcMembers
-public class DDTracer: DatadogObjc.OTTracer {
+public class DDTracer: NSObject, DatadogObjc.OTTracer {
```

Comment:
@buranmert this change of adding `NSObject` subclass for `DDTracer` fixed critical issue with DatadogObjc tracing. Without `NSObject` subclassing the entire `DDTracer` was not visible to Objective-C code (⚠️).

It means that `OTGlobal+objc` API was impossible to use 😞 as `DDTracer` was never visible. Because of this, IMO we can just remove it without deprecating, WDYT?

Reply:
oh that's bad, then yes we can just remove it 👍 
as i said in Slack: we are not really testing generated ObjC interface, we should be calling our SDK from ObjC somehow in tests to ensure that, wdyt?

Reply:
This would be cool and would help us avoid such mistakes as this one. But it seems difficult at the first glance to build Objc unit tests, as the assertions would need to call non-bridged Swift code. Anyway, I think we should consider this in `RUMM-896` 👌.